File: lintcode/remove_duplicates_from_sorted_arrayII.py
class Solution:
    """
    @param A: a list of integers
    @return an integer
    """
    def removeDuplicates(self, A):
        current = 0
        for num in A:
            if current < 2 or num > A[current - 2]:
                A[current] = num
                current += 1
        return current

File: leetcode/remove_duplicates_from_sorted_arrayII.js
/* Follow up for "Remove Duplicates":
 * What if duplicates are allowed at most twice?
 * Time complexity: O(n)
 * Space complexity: O(1)
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    for (var current = 0, i = 0; i < nums.length; ++i) {
        if (current < 2 || nums[i] > nums[current - 2]) {
            nums[current++] = num
        }
    }    
    return current
};
